synapseclient/core/credentials/cred_data.py

```diff
@@ -103,14 +105,48 @@ class SynapseAuthTokenCredentials(SynapseCredentials):
     def get_keyring_service_name(cls):
         return 'SYNAPSE.ORG_CLIENT_AUTH_TOKEN'
 
-    def __init__(self, username, token):
-        self._username = username
+    @classmethod
+    def _validate_token(cls, token):
+        # decode the token to ensure it minimally has view scope.
+        # if it doesn't raise an error, the client will not be useful without it.
+
+        # if for any reason we are not able to decode the token and check its scopes
+        # we do NOT raise an error. this is to accommodate the possibility of a changed
+        # token format some day that this version of the client may still be able to
+        # pass as a bearer token.
+        try:
+            token_body = json.loads(
+                str(
+                    base64.urlsafe_b64decode(
+                        token.split('.')[1] + '=='
```

Comment:
do we have to append the padding `==`

Reply:
Added an explanation. We're basically always adding padding because Python b64decode will ignore any unnecessary padding so adding the maximum padding doesn't hurt (but we do need to add the padding as the tokens we get from the backend are unpadded). https://stackoverflow.com/a/49459036